tests/system/src/test/java/test/javafx/scene/InitialNodesMemoryLeakTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.scene.Group;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.stage.WindowEvent;
+import junit.framework.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import test.util.Util;
+
+import java.lang.ref.WeakReference;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.fail;
+
+public class InitialNodesMemoryLeakTest {
+
+    static CountDownLatch startupLatch;
+    static WeakReference<Group> toCleanUp;
+
+    public static class TestApp extends Application {
+
+        @Override
+        public void start(Stage stage) throws Exception {
+            toCleanUp = new WeakReference<>(new Group());
+
+            Group root = new Group(toCleanUp.get());
```

Comment:
There is a very small (but non-zero) window where the newly created child Group could be garbage collected between the time it goes out of scope (right after adding it as the referent of `toCleanUp`) and the call to `toCleanUp().get`. I recommend holding the new Group a local variable temporarily. Something like this:

```
    Group group = new Group();
    toCleanUp = new WeakReference<>(group);
    Group root = new Group(group);
```

You could also set `group = null` after creating the` root`, if you want, but it isn't necessary, since `group` will go out of scope when `start()` returns, leaving the children list of `root` as the only hard reference to the object (until you clear it).

Reply:
That's a great Idea! I've changed the code accordingly.